### DISCUSSION TITLE
[vcpkg/host-triplet] Make the host triplet selection better

### DISCUSSION THREAD

**Discussion by Neumann-A** [microsoft/vcpkg#21597] [Ideas]
Problem: `x64-windows-static` is not a host triplet on `x64 & windows`
Observed while by trying to make qtbase depend on itself for the host.

vcpkg's selection of the host triplet is quite rudimentary in terms of the host triplet selection. This affects **all** custom triplets!

Suggestion:
Look into the target triplet (`VCPKG_SYSTEM_NAME` & `VCPKG_TARGET_ARCHITECTURE`) and compare the values against the host. 
- If the values match the host use the target triplet as the host triplet.
- if not: do as before.
- if user passes: --host-triplet. Don't do anything and just use the provided value (current behavior) 

cc @ras0219-msft @strega-nil 

other observation: Compiler detection seems to run with host = target triplet according to `--debug` (Intended or bug?)

**Comment by Hoikas** [comment]
FWIW, if you need a host triplet copy of python3 and vcpkg selects x64-windows-static instead of x64-windows, then the built python.exe will be unable to load extension modules.

**Comment by Neumann-A** [comment]
Isn't python handled by `vcpkg_find_acquire_program` any way ?

**Comment by Hoikas** [comment]
> Isn't python handled by vcpkg_find_acquire_program any way ?

At this time, yes. However, if you want a functioning Python 3 interpreter on Windows 7 for your portfile, then you would need to install python3[deprecated-win7-support] for the host. Currently, no one does that, but I think it's worth keeping in mind that some ports have major caveats with certain triplets.

**Comment by autoantwort** [comment]
This only works if people don't set `VCPKG_SYSTEM_NAME` & `VCPKG_TARGET_ARCHITECTURE` depending on the port name (afaik currently allowed)
